packages/itwinui-css/src/button/button.scss

```diff
@@ -33,10 +33,11 @@
 }
 
 @mixin iui-button-icon {
-  @include iui-icon-style('m');
-  transition: fill var(--iui-duration-1) ease-out;
-  fill: currentColor;
-  margin-block: var(--iui-size-2xs);
+  > * {
```

Comment:
I think this should target all svg descendants instead of just the direct child.

```suggestion
  svg {
```

This is necessary for notification marker to work in react with the following dom structure:

```html
<button>
  <span class='iui-button-icon'>
    <span class='iui-notification-marker'>
      <svg>...</svg>
```

Reply:
I had it as `svg` first, but then thought that since `.iui-button-icon` was styling let's say tag `X`, and we are now wrapping `X` with a span with `.iui-button-icon`, the CSS style should style `> *` to keep the styling the same.

For nested svgs like in the example you gave, I think the properties would be inherited down the subtree to the SVG. I tested it out in React and it seems to work:

![image](https://user-images.githubusercontent.com/45748283/198715965-1c406c2b-4618-49c1-98a1-1d6c9443960a.png)

Reply:
The styles tab will show all properties from every ancestor all the way up the tree if you scroll up. It is not an accurate representation of which styles are actually taking effect (Computed tab is more reliable for that). But I'm pretty sure `fill` is the only property out of all of those that gets inherited. The rest stay on the parent span.  The fact that it "works" is just a coincidence.

I'm not sure what you mean by "tag X" or how it would "break" if we target all descendant svgs. The `iui-button-icon` has one job: style svgs inside it.

Reply:
**[DONE]** Sounds good, yes, I think you're right. Not all attributes were inherited. So we need to style the svg directly instead of relying on inheritance. I implemented your suggested change.

> I'm not sure what you mean by "tag X"

By tag `X` I meant any arbitrary tag

> ... what you mean by "tag X" or how it would "break" if we target all descendant svgs

I didn't mean it would break anything but I felt that since we were not explicitly styling `svg` before, now also there is no need to explicitly style `svg`. That's why I styled `> *` instead of `svg`.